src/domain/Tables/Table/subcomponents/TableRow.tsx

```diff
@@ -267,85 +214,89 @@ const TableRow = <T extends {}>(props: ITableRowProps<T>) => {
     actions
   } = row
 
+  const previousRowProps = usePrevious<IRowProps<T>>(row)
   const [movement, setMovement] = useState<number>(0)
-  const [previousProgress, setPreviousProgress] = useState<number>(0)
   const [hasHovered, setHasHovered] = useState<boolean>(false)
-  const [key, setKey] = useState<string>('')
 
+  const isSelected = isSelectable && propsIsSelected && hasLeftAction
   const setHoveredTrue = useCallback(() => setHasHovered(true), [setHasHovered])
   const setHoveredFalse = useCallback(() => setHasHovered(false), [setHasHovered])
-  const hasSwipeActions = interactionType === Table.InteractionType.Swipe && !!actions && actions.length > 0
-  const hasHoverActions = interactionType === Table.InteractionType.Hover && !!actions && actions.length > 0
+  const hasSwipeActions = interactionType === InteractionType.Swipe && !!actions && actions.length > 0
+  const hasHoverActions = interactionType === InteractionType.Hover && !!actions && actions.length > 0
+  const hasHoverButton = (!isSelected && hasHoverActions) ? hasHovered : false
 
   const swipeContentWidth = hasSwipeActions && actions ? getIconButtonWidth(actions) + (actions.length - 1) * Variables.Spacing.sXSmall + 2 * Variables.Spacing.sMedium : 0
 
   const onDragBlind = useDrag((useDragProps: IUseDragProps) => {
-    if (hasSwipeActions && useDragProps._movement[0] <= movement && useDragProps._movement[0] !== 0) {
-      if (Math.floor(useDragProps._movement[0]) !== movement) {
-        setMovement(Math.min(Math.trunc(Math.abs(useDragProps._movement[0])), swipeContentWidth))
+    if (hasSwipeActions && useDragProps._movement[0] !== 0) {
+      // new position = oldposition - movement, clamped by [0, maxwidth]
+      let newMovement = Math.floor(movement - useDragProps._movement[0])
+      newMovement = Math.max(Math.min(newMovement, swipeContentWidth), 0)
+
+      // When dragging stops, snap to one of the two ends
+      if (!useDragProps.dragging) {
+        newMovement = (newMovement < (swipeContentWidth / 2)) ? 0 : swipeContentWidth
       }
 
-      if (!useDragProps.dragging && useDragProps._movement[0] < 0) {
-        setMovement((Math.abs(useDragProps._movement[0]) < (swipeContentWidth / 2)) ? 0 : swipeContentWidth)
+      if (newMovement !== movement) {
+        setMovement(newMovement)
+
+        if (newMovement > 0) {
+          setExpandedSwipeCellRow(id)
+        } else if (newMovement === 0 && expandedSwipeCellRow !== id) {
+          setExpandedSwipeCellRow(null)
+        }
       }
     }
```

Comment:
Rewrote this logic
 1. Originally it had a bug where swipes to the right would not always snap to 0 when they should've. I rewrote the movement to make more sense which fixes it.
2. `setExpandedSwipeCellRow` in here rather than using an effect is much more efficient and avoids a render cycle.